include/index_config.h

```diff
@@ -160,15 +157,31 @@ class IndexConfigBuilder
         return *this;
     }
 
-    IndexConfigBuilder &with_search_threads(uint32_t search_threads)
+    IndexConfigBuilder &with_index_write_params(std::shared_ptr<IndexWriteParameters> index_write_params_ptr)
+    {
+        if (index_write_params_ptr == nullptr)
+        {
+            diskann::cout << "Passed, empty build_params while creating index config" << std::endl;
+            return *this;
+        }
+        this->_index_write_params = index_write_params_ptr;
+        return *this;
+    }
+
+    IndexConfigBuilder &with_index_search_params(IndexSearchParams &search_params)
     {
-        this->_search_threads = search_threads;
+        this->_index_search_params = std::make_shared<IndexSearchParams>(search_params);
```

Comment:
Doesn't make_shared<> require a pointer?

Reply:
It is a shared pointer, stored inside index_config